tests/cupy_tests/core_tests/test_raw.py

```diff
@@ -78,13 +83,41 @@
 }
 '''
 
+if 'CUPY_CACHE_DIR' in os.environ:
+    _old_cache_dir = os.environ['CUPY_CACHE_DIR']
+    _is_cache_env_var_set = True
+else:
+    _old_cache_dir = os.path.expanduser('~/.cupy/kernel_cache')
+    _is_cache_env_var_set = False
+_test_cache_dir = None
 
+
+@testing.parameterize(*testing.product({
+    'backend': ('nvrtc', 'nvcc'),
+}))
 class TestRaw(unittest.TestCase):
 
     def setUp(self):
-        self.kern = cupy.RawKernel(_test_source1, 'test_sum')
-        self.mod2 = cupy.RawModule(_test_source2)
-        self.mod3 = cupy.RawModule(_test_source3, ("-DPRECISION=2",))
+        global _test_cache_dir
+        _test_cache_dir = mkdtemp()
+        os.environ['CUPY_CACHE_DIR'] = _test_cache_dir
+
+        self.kern = cupy.RawKernel(_test_source1, 'test_sum',
+                                   backend=self.backend)
+        self.mod2 = cupy.RawModule(_test_source2, backend=self.backend)
+        self.mod3 = cupy.RawModule(_test_source3, ("-DPRECISION=2",),
```

Comment:
Let's use single quotes for consistency (even though this code has existed from before).

```suggestion
        self.mod3 = cupy.RawModule(_test_source3, ('-DPRECISION=2',),
```